### PR TITLE
[PT2 Dynamo Hackathon] Fix simple bug in inline dict

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103187

Fixes: https://github.com/pytorch/pytorch/issues/101980

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

### CODE REVIEW

File: test/dynamo/test_misc.py [hunk 1]
```diff
@@ -5353,6 +5353,16 @@ def test_hash_getitem_slice(self):
         self.assertTrue(s == s2)
         self.assertTrue(s != s3)
 
+    def test_inline_dict_function(self):
+        def _result_type_dict(dtype):
+            return {bool: torch.float32}[dtype]
+
```
Comment (yanboliang) [inline test/dynamo/test_misc.py:5359]: It's better we can add another scenario where the dict is passed in as an argument:
```
@torch.compile()
def fn(d, x):
    if d[x] is torch.float32:
        return d[x]
    else:
        ...

dd = {bool: torch.float32}
fn(dd, bool)
```

You can refer this PR: https://github.com/pytorch/pytorch/pull/93026

Comment (tugsbayasgalan) [reply test/dynamo/test_misc.py:5359]: ok will do

Comment (tugsbayasgalan) [reply test/dynamo/test_misc.py:5359]: done

**Review by yanboliang — APPROVED** [review]

**Pull request closed without merging**